src/ui_context.rs

```diff
@@ -200,12 +196,9 @@ impl UIContext {
             *cookie = None;
         }
         if let Ok(mut signal_handler_id) = MOUSE_NOTIFY_SIGNAL_ID.lock() {
-            if let Some(handler) = *signal_handler_id {
-                unsafe {
-                    gobject_sys::g_signal_handler_disconnect(window.to_glib_none().0, handler);
-                }
+            if let Some(handler) = signal_handler_id.take() {
+                window.disconnect(handler);
```

Comment:
Oh is this new? Last I checked I had to use gobject-sys for signals disconnection.

Reply:
No, it always was like that. But `Option::take()` is a very under-appreciated function :)